server/src/main/java/org/opensearch/index/compositeindex/Dimension.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.compositeindex;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+/**
+ * Composite index dimension base class
+ *
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public class Dimension implements ToXContent {
+    private final String field;
+
+    public Dimension(String field) {
+        this.field = field;
+    }
+
+    public String getField() {
+        return field;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
```

Comment:
Do we need to have a `static fromXContent` method for all of these as well?

Reply:
`fromXContent` is not needed as we parse the user input and map to objects in the `StarTreeMapper`. There are lots of validations as we parse the dimensions and metrics and its in line with other mappers, so will follow the same for now.

`toXContent` is needed as part of merge flow as all the properties are merged to a json and this helps in structuring the objects back to the same way we read them with the additional type parameter since we lack in merge flow ( so initially as part of user input, the mapperBuilders object will define the dimension type, later as part of merge flow, the builders will be null, so we are adding type when we do `toXContent` )